api/main.py

```diff
@@ -45,6 +46,18 @@ def semantic():
 # ------------------------------------ chat ------------------------------------
 
 
+def clean_history(history: list[Message]) -> list[Message]:
```

Comment:
might as well attempt to fix things if they're sent incorrectly

Reply:
it's not any more - there was a frontend bug (or 5) which was causing this. But it seems like a good default to have anyway